libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py

```
@@ -395,6 +395,19 @@ def check_migration_timeout_suspend(params):
         graphics_uri = libvirt_vm.complete_uri(migrate_dest_ip, "spice", graphics_port)
         extra = "--graphicsuri %s" % graphics_uri
 
+    # retrieve shared storage image path from existing params
+    try:
+        nfs_mount_path = params.get("nfs_mount_dir")
+        image = params.get("image_name").split("/")[-1]
+        image_format = params.get("image_format")
+        image = "%s.%s" % (image, image_format)
+        params["migrate_shared_storage"] = os.path.join(nfs_mount_path, image)
+        logging.debug("shared storage image location: %s",
+                      params["migrate_shared_storage"])
+    except IndexError:
+        # it will be taken care by default image jeos-23-64
```

can you keep the older implementation for backward compatibility, instead user wants to use the given path,

if we want to use the manipulated path like this lets us set the param `migrate_shared_storage = "" `and check if none then lets do this try otherwise silently use the user defined value?